src/tasks/netcore/NetCoreTaskHelper.ts

```diff
@@ -0,0 +1,308 @@
+        const buildOptions = buildDefinition.dockerBuild;
+        const helperOptions = buildDefinition.netCore || {};
+
+        helperOptions.appProject = await NetCoreTaskHelper.inferAppProject(context.folder, helperOptions); // This method internally checks the user-defined input first
+
+        const appName = await NetCoreTaskHelper.inferAppName(context.folder, helperOptions);
+
+        // tslint:disable: no-invalid-template-strings
+        buildOptions.context = buildOptions.context || '${workspaceFolder}';
+        buildOptions.dockerfile = buildOptions.dockerfile || path.join('${workspaceFolder}', 'Dockerfile');
+        // tslint:enable: no-invalid-template-strings
+        buildOptions.tag = buildOptions.tag || await this.inferImage(appName, buildDefinition);
+        buildOptions.labels = buildOptions.labels || NetCoreTaskHelper.defaultLabels;
+
+        return buildOptions;
+    }
+
+    public async resolveDockerRunOptions(context: DockerRunTaskContext, runDefinition: NetCoreRunTaskDefinition): Promise<DockerRunOptions> {
+        const runOptions = runDefinition.dockerRun;
+        const helperOptions = runDefinition.netCore || {};
+
+        helperOptions.appProject = await NetCoreTaskHelper.inferAppProject(context.folder, helperOptions); // This method internally checks the user-defined input first
+
+        const appName = await NetCoreTaskHelper.inferAppName(context.folder, helperOptions);
+
+        runOptions.containerName = runOptions.containerName || `${appName}-dev`;
+        runOptions.labels = runOptions.labels || NetCoreTaskHelper.defaultLabels;
+        runOptions.os = runOptions.os || 'Linux';
+        runOptions.image = runOptions.image || await this.inferImage(appName, context.buildDefinition);
+
+        const ssl = helperOptions.configureSsl !== undefined ? helperOptions.configureSsl : await NetCoreTaskHelper.inferSsl(context.folder, helperOptions);
+        const userSecrets = ssl === true ? true : await this.inferUserSecrets(context.folder, helperOptions);
+
+        if (userSecrets) {
+            runOptions.env = runOptions.env || {};
+            runOptions.env.ASPNETCORE_ENVIRONMENT = runOptions.env.ASPNETCORE_ENVIRONMENT || 'Development';
+
+            if (ssl) {
+                // tslint:disable-next-line: no-http-string
+                runOptions.env.ASPNETCORE_URLS = runOptions.env.ASPNETCORE_URLS || 'https://+:443;http://+:80';
+            }
+        }
+
+        runOptions.volumes = await this.inferVolumes(context.folder, runOptions, helperOptions, ssl, userSecrets); // Volumes specifically are unioned with the user input (their input does not override except where the container path is the same)
+
+        return runOptions;
+    }
+
+    public static async inferAppName(folder: WorkspaceFolder, helperOptions: NetCoreTaskOptions | NetCoreDebugOptions): Promise<string> {

```

Comment:
>inferAppName [](start = 24, length = 12)

Contra this function's name, its purpose seems to be to generate a valid image name from a given project file name.  But it's also used to generate a container name from the project file name.

Why not separate these concerns into three functions createValidImageName, createValidContainerName, getProjectName?

(I also object to "AppName" being used instead of "ProjectName" as AppName doesn't correspond to an established concept in C# and will probably be confused for the OutputAssemblyName).

Reply:
I created a getImageName and getContainerName, and consequently there's no need at all of an appName.